source/plugins/go/src/utils.go

```diff
@@ -259,3 +261,119 @@ func convertMsgPackEntriesToMsgpBytes(fluentForwardTag string, msgPackEntries []
 
 	return msgpBytes
 }
+
+// includes files in subdirectories
+// TODO: consider replacing this with a iterator of some sort, constructing the map of
+// files all at once might use more memory than necessary
+func GetSizeOfAllFilesInDir(root_dir string) map[string]int64 {
+	output_map := make(map[string]int64)
+	getSizeOfAllFilesInDirImpl(&root_dir, "", &output_map)
+	return output_map
+}
+
+func getSizeOfAllFilesInDirImpl(root_dir *string, preceeding_dir string, storage_dict *map[string]int64) {
+	// container_full_path := filepath.Join(preceeding_dir_segments)
+	files_and_folders, err := ioutil.ReadDir(filepath.Join(*root_dir, preceeding_dir))
+	if err != nil {
+		Log("ERROR: reading dir " + err.Error())
```

Comment:
we can just return if there is any error for ioutil.ReadDir rather proceeding further, right?

Reply:
Good point, fixing that